x/gov/handler_test.go

```diff
@@ -814,7 +776,8 @@ func TestTally(t *testing.T) {
 		WantCheckErr   *errors.Error
 		WantDeliverErr *errors.Error
 		ExpResult      Proposal_Result
-		PostChecks     func(t *testing.T, kvStore weave.KVStore)
+		PostChecks     func(t *testing.T, db weave.KVStore)
```

Comment:
This is getting more and more complex, and honestly it is hard to follow.

I would extract the for loop into a few helpers and then turn each item in the table into a separate function to test. Then one can read the logic without jumping constantly between the testcase and the for loop.